ndsplines/ndsplines.py

```diff
@@ -45,41 +45,52 @@ class BSplineNDInterpolator(object):
     def __init__(self, knots, coefficients, orders, periodic=False, extrapolate=True):
         self.knots = knots
         self.xdim = len(knots) # dimension of knots
-
-        self.coefficients = coefficients
-        if coefficients.ndim == self.xdim:
-            self.squeeze = True
-            self.coefficients = coefficients[None, ...]
-        else:
-            self.coefficients = coefficients
-            self.squeeze = False
-
-        self.ydim = self.coefficients.shape[0] # dimension of coefficeints
-
+        self.xshape = tuple(knot.size for knot in knots)
+        
         self.orders = np.broadcast_to(orders, (self.xdim,))
+        self.max_order = np.max(self.orders)
         self.periodic = np.broadcast_to(periodic, (self.xdim,))
         self.extrapolate = np.broadcast_to(extrapolate, (self.xdim,2))
 
-        self.coefficient_op = [0,] + list(i for i in range(2,self.xdim+2)) + [1,]
-        self.u_ops = [[1, i+2] for i in range(self.xdim)]
-        self.output_op = [0,1]
+        assert np.all(coefficients.shape[:self.xdim] == self.xshape - self.orders - 1)
```

Comment:
Isn't it generally favorable to raise an exception? That way you give the user a chance to catch it (probably unlikely in this case) and/or print a more helpful message (i.e. the shape(s) that were expected vs received).

Reply:
I was wondering about the convention, I'll fix it.